File: 1894-merge-strings-alternately/merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        '''
        iterations
        when l2 or l1 ends then append rest of string
        '''
        l1 = len(word1)
        l2 = len(word2)
        ans = ''
        for i in range(0,min(l1,l2)):
                ans += word1[i] +word2[i]
        
        if l1 > l2:
            ans += word1[l2:]
        else:
            ans += word2[l1:]
        return ans

        

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {

        int l1 = word1.length();
        int l2 = word2.length();
        int l = Math.min(l1,l2);

        StringBuilder merged = new StringBuilder();

        for(int i=0;i<l;i++){
            merged.append(word1.charAt(i));
            merged.append(word2.charAt(i));
        }

        if(l1 > l2){
            merged.append(word1.substring(l));
        }
        else{
            merged.append(word2.substring(l));
        }

        return merged.toString();
    }
}